iceoryx_posh/test/moduletests/test_popo_publisher_port.cpp

```diff
@@ -72,51 +73,75 @@ class PublisherPort_test : public Test
+    iox::popo::PublisherOptions m_noOfferOnCreatePublisherOptions{0U, "", false};
+    ;
 
-    // publisher port w/o history
+    // publisher port w/o offer on create
     iox::popo::PublisherPortData m_publisherPortData{
-        iox::capro::ServiceDescription("a", "b", "c"), "myApp", &m_memoryManager, m_publisherOptions};
-    iox::popo::PublisherPortRouDi m_sutRouDiSide{&m_publisherPortData};
-    iox::popo::PublisherPortUser m_sutUserSide{&m_publisherPortData};
+        iox::capro::ServiceDescription("a", "b", "c"), "myApp", &m_memoryManager, m_noOfferOnCreatePublisherOptions};
+    iox::popo::PublisherPortRouDi m_sutNoOfferOnCreateRouDiSide{&m_publisherPortData};
+    iox::popo::PublisherPortUser m_sutNoOfferOnCreateUserSide{&m_publisherPortData};
 
     // publisher port w/ history
-    iox::popo::PublisherOptions m_publisherPortOptions{iox::MAX_PUBLISHER_HISTORY};
+    iox::popo::PublisherOptions m_withHistoryPublisherOptions{iox::MAX_PUBLISHER_HISTORY, "", true};
     iox::popo::PublisherPortData m_publisherPortDataHistory{
-        iox::capro::ServiceDescription("x", "y", "z"), "myApp", &m_memoryManager, m_publisherPortOptions};
-    iox::popo::PublisherPortUser m_sutWithHistoryUseriSide{&m_publisherPortDataHistory};
+        iox::capro::ServiceDescription("x", "y", "z"), "myApp", &m_memoryManager, m_withHistoryPublisherOptions};
+    iox::popo::PublisherPortUser m_sutWithHistoryUserSide{&m_publisherPortDataHistory};
     iox::popo::PublisherPortRouDi m_sutWithHistoryRouDiSide{&m_publisherPortDataHistory};
+
+    // publisher port w/ default options
+    iox::popo::PublisherOptions m_withDefaultPublisherOptions{};
+    iox::popo::PublisherPortData m_publisherPortDataDefault{
+        iox::capro::ServiceDescription("x", "y", "z"), "myApp", &m_memoryManager, m_withDefaultPublisherOptions};
+    iox::popo::PublisherPortUser m_sutWithDefaultOptionsUseriSide{&m_publisherPortDataDefault};
+    iox::popo::PublisherPortRouDi m_sutWithDefaultOptionsRouDiSide{&m_publisherPortDataDefault};
 };
 
-TEST_F(PublisherPort_test, initialStateIsNotOffered)
+TEST_F(PublisherPort_test, initialStateIsOfferedWithDefaultOptions)
+{
+    EXPECT_TRUE(m_sutWithDefaultOptionsUseriSide.isOffered());
+}
+
+TEST_F(PublisherPort_test, initialStateIsNotOfferedWhenNoOfferOnCreate)
 {
-    EXPECT_FALSE(m_sutUserSide.isOffered());
+    EXPECT_FALSE(m_sutNoOfferOnCreateUserSide.isOffered());
 }
 
+
 TEST_F(PublisherPort_test, initialStateIsNoSubscribers)
 {
-    EXPECT_FALSE(m_sutUserSide.hasSubscribers());
+    EXPECT_FALSE(m_sutNoOfferOnCreateUserSide.hasSubscribers());
 }
 
-TEST_F(PublisherPort_test, initialStateReturnsNoCaProMessage)
+TEST_F(PublisherPort_test, initialStateReturnsOfferCaProMessageWithDefaultOptions)
 {
-    auto maybeCaproMessage = m_sutRouDiSide.tryGetCaProMessage();
+    auto maybeCaproMessage = m_sutWithDefaultOptionsRouDiSide.tryGetCaProMessage();
+
+    EXPECT_TRUE(maybeCaproMessage.has_value());
```

Comment:
Please use ``ASSERT_TRUE``, otherwise there might be undefined behavior in the next line.